corehq/apps/domain/views.py

```diff
@@ -3168,25 +3168,28 @@ def update_dataset_map(instance, dict_):
                     value = [DataValueMap(**v) for v in value]
                 instance[key] = value
 
-        dataset_maps = json.loads(request.POST['dataset_maps'])
-        current_dataset_maps = get_dataset_maps(request.domain)
-        i = 0
-        for i, dataset_map in enumerate(current_dataset_maps):
-            if i < len(dataset_maps):
-                # Update current dataset maps
-                update_dataset_map(dataset_map, dataset_maps[i])
-                dataset_map.save()
-            else:
-                # Delete removed dataset maps
-                dataset_map.delete()
-        if (i + 1) < len(dataset_maps):
-            # Insert new dataset maps
-            for j in range(i + 1, len(dataset_maps)):
-                dataset_map = DataSetMap(domain=request.domain)
-                update_dataset_map(dataset_map, dataset_maps[j])
-                dataset_map.save()
-        get_dataset_maps.clear(request.domain)
-        return HttpResponse(_('DHIS2 DataSet Maps saved'))
+        try:
+            new_dataset_maps = json.loads(request.POST['dataset_maps'])
+            current_dataset_maps = get_dataset_maps(request.domain)
+            i = -1
+            for i, dataset_map in enumerate(current_dataset_maps):
+                if i < len(new_dataset_maps):
+                    # Update current dataset maps
+                    update_dataset_map(dataset_map, new_dataset_maps[i])
+                    dataset_map.save()
+                else:
+                    # Delete removed dataset maps
+                    dataset_map.delete()
+            if i + 1 < len(new_dataset_maps):
+                # Insert new dataset maps
+                for j in range(i + 1, len(new_dataset_maps)):
+                    dataset_map = DataSetMap(domain=request.domain)
+                    update_dataset_map(dataset_map, new_dataset_maps[j])
+                    dataset_map.save()
+            get_dataset_maps.clear(request.domain)
+            return json_response({'success': _('DHIS2 DataSet Maps saved')})
+        except Exception as err:
```

Comment:
The exception I was looking for was from trying to save a string as an integer. It was resolved by commit "Cast dayOfMonth as a number" (f4ac8c3). I'm not expecting exceptions in the future, but I figured it would be useful to leave this in, at least until we have one non-demo integration up and running.